helix-term/src/commands.rs

```diff
@@ -3002,11 +3002,12 @@ fn goto_prev_diag(cx: &mut Context) {
         .cursor(doc.text().slice(..));
 
     let diag = doc
-        .diagnostics()
-        .iter()
+        .shown_diagnostics()
+        .collect::<Vec<_>>()
+        .into_iter()
         .rev()
```

Comment:
You don't need to collect here if you add a double ended iterator bound (see my other comment)